internal/functions/coviddata/get-covid-data.go

```diff
@@ -92,13 +85,70 @@ func GetCovidData(w http.ResponseWriter, r *http.Request) error {
 
 	logger.Infof("fetched data: %+v", increase)
 
-	totalsData := totals
+	return &increase, nil
+}
+
+// GetCovidData handler.
+func GetCovidData(w http.ResponseWriter, r *http.Request) {
 
+	var ctx = r.Context()
+	logger := logging.FromContext(ctx)
+	client := store.Client{}
+
+	var req getRequest
+
+	if !httputils.DecodeJSONOrReportError(w, r, &req) {
+		return
+	}
+
+	date := req.Date
+
+	// if no date was specified in input
+	// and there is no data for today, try to get
+	// data for yesterday
+	shouldRetry := false
+
+	if date == "" {
+		date = utils.GetTimeNow().Format("20060102")
+		shouldRetry = true
+	}
+
+	failed := false
+
+	totalsData, err := fetchTotals(ctx, client, date)
 	if err != nil {
-		panic(fmt.Sprintf("could not convert firestore fields to data: %s", err))
+		logger.Errorf("Error fetching data from firestore: %v", err)
+		failed = true
+	}
+	increaseData, err := fetchIncrease(ctx, client, date)
+	if err != nil {
+		logger.Errorf("Error fetching data from firestore: %v", err)
+		failed = true
 	}
 
-	increaseData := increase
+	if failed && !shouldRetry {
+		httputils.SendErrorResponse(w, r, rpccode.Code_INTERNAL, "Unknown error")
+		return
+	}
+
+	if failed && shouldRetry {
```

Comment:
I think this is more _fallback_ than _retry_, isn't it? :thinking:  
Anyway, I think these two blocks are so similar that they could be wrapped into sth like `func fetchData(date string) ((data, data), error)`.

Reply:
right! I'll rename it to fallback, good point.

They are very similar, it is definitely boilerplate, though I'll have to come up with a better name, we already have fetchData in the same package :))

Reply:
`fetchAll`? Whatever :smile:

Reply:
done